### PR TITLE
New indicator

### LINKED ISSUE #22: Badge in the proposal list when user has already cast a vote
It has been proposed to add a badge in the top-right corner of the proposal cards in the main screen to indicate that a vote has been cast on the corresponding proposal.

The solution of this issue will be similar to #23

### PR DESCRIPTION
**Pull request by emieldesmidt**
Added a subtle indicator that a vote proposal has not recieved your vote yet.

Closes #22

image

### CODE REVIEW

File: voting/src/main/java/nl/tudelft/trustchain/voting/blockListAdapter.kt [hunk 1]
```diff
@@ -48,23 +50,35 @@ class blockListAdapter(
...
+            JSONObject(proposalBlock.transaction["message"].toString()).get("VOTE_SUBJECT")
+                .toString()
+
+        holder.propDate.text =
+            DateFormat.format("EEE MMM d HH:mm", proposalBlock.timestamp).toString()
+
         holder.propTitle.text =
-            JSONObject(myDataset[position].transaction["message"].toString()).get("VOTE_SUBJECT")
+            JSONObject(proposalBlock.transaction["message"].toString()).get("VOTE_SUBJECT")
                 .toString()
```
Comment (PraveshMoelchand) [inline voting/src/main/java/nl/tudelft/trustchain/voting/blockListAdapter.kt:66]: Isn't this duplicate with lines 57-59?

**Comment by KoningR** [comment]
Due to two odd behaviors, the 'new' message can become quite confusing:
- 'new' does not disappear directly after vote has been cast, only after refreshing the screen.
- 'new' appears on votes that I have not voted upon, but where voting has been completed already.

**Comment by PraveshMoelchand** [comment]
> * 'new' does not disappear directly after vote has been cast, only after refreshing the screen.

I have not encountered this behaviour while testing

**Comment by emieldesmidt** [comment]
> > * 'new' does not disappear directly after vote has been cast, only after refreshing the screen.
> 
> I have not encountered this behaviour while testing

Same for me, are you sure @KoningR ?

**Comment by emieldesmidt** [comment]
Update, the new tag can get pushed off screen when the proposal wraps around. Will fix.
image

**Comment by erwinvanthiel** [comment]
As @KoningR already mentioned "'new' appears on votes that I have not voted upon, but where voting has been completed already." This can be considered confusing in my opinion but it's not really a problem for me. Another thing is that the new tag lags when scrolling, so it looks like its flickering on all proposal blocks. When you stop scrolling it does restore itself so its neither really a big problem but it did confuse me a bit as well when I saw it for the first time.

**Comment by PraveshMoelchand** [comment]
I have not encountered the flickering `new` text.

I do agree that the `new` badge should change when voting has been completed. Working on that now

**Comment by PraveshMoelchand** [comment]
Implemented the stuff:
New proposal in case of no votes and only votes by others:
afbeelding
afbeelding

Voted on proposal, but not yet complete:
afbeelding

Completed proposal:
afbeelding

**Comment by PraveshMoelchand** [comment]
I know, the margin between the title and indicator seems a big in some cases, but this is because words are not broken up. Somehow, Android doesnt see the space that is created by moving the words to the next line instead of breaking them up. See:
afbeelding

**Pull request merged**